src/modules/reducers/submittedTransactions.js

```diff
@@ -52,11 +61,19 @@ export const updateTx = tx => {
     };
 };
 
-export const dismissTx = txHash => {
+export const dismissTx = (txHash, dismissState) => {
     return async dispatch => {
         try {
             const transactions = Object.assign({}, store.getState().submittedTransactions.transactions);
-            transactions[txHash].isDismissed = true;
+            if (txHash === undefined) {
+                Object.keys(transactions).forEach(key => {
+                    transactions[key].isDismissed =
```

Comment:
If this logic is needed, it may be more readable:
```
if (transactions[key].isDismissed !== "archive") {
  transactions[key].isDismissed = dismissState;
}
```